Test/DataAccessLayer/Repositories/StudentsRepository.cs

```diff
@@ -0,0 +1,73 @@
+﻿using BusinessLayer.Models;
+
+namespace DataAccessLayer.Repositories
+{
+    public class StudentsRepository : IStudentsRepository
+    {
+        //get all students
+        public async Task<IEnumerable<Student>>  GetStudents()
+        {
+            return await DataSource();
+        }
+
+        //get by id
+        public async Task<Student> GetStudentByID(int id)
+        {
+            List<Student> students = await DataSource();
+            if (!students.Exists(student => student.Id == id))
+            {
+                throw new Exception("Nu exista student!");
```

Comment:
please use a better error handling mechanism here. also, the error messages should be written in English.

*to be applied everywhere it's needed in this file